aws/sdk-codegen/src/main/kotlin/software/amazon/smithy/rustsdk/customize/s3/S3Decorator.kt

```diff
@@ -48,6 +55,12 @@ class S3Decorator : RustCodegenDecorator<ClientProtocolGenerator, ClientCodegenC
         ShapeId.from("com.amazonaws.s3#GetObjectAttributesOutput"),
     )
 
+    private val serializeZeroValuesAllowList = setOf(
+        // Omitting this when zero would change the meaning of the range.
+        // https://docs.aws.amazon.com/AmazonS3/latest/API/API_ScanRange.html
+        ShapeId.from("com.amazonaws.s3#Start"),
```

Comment:
I believe `End` should also accept zero.

https://docs.aws.amazon.com/AmazonS3/latest/API/API_ScanRange.html#AmazonS3-Type-ScanRange-End
```bash
[~] aws s3api select-object-content \
--bucket ${BUCKET_NAME_OMMITED} \
--key data-1m.json \
--expression 'SELECT * FROM s3object s' \
--expression-type SQL \
--input-serialization '{"JSON": {"Type": "LINES"}, "CompressionType": "NONE"}' \
--output-serialization '{"JSON": {}}' \
--scan-range Start=0,End=0 \
0.json
[~] echo $?
0
[~] cat 0.json
{"a":"foo","b":1,"c":"aaa"}
```